src/lib/utils.ts

```diff
@@ -4,6 +4,19 @@ import * as ipaddr from 'ipaddr.js';
 import { promisify } from 'util';
 import { delay } from 'bluebird';
 
+// process.hrtime() will give nanos, but it is from an unknown relative time, not epoch.
+// This approach calculates the difference and adds to get the current nano time.
+const loadNs = process.hrtime();
+const loadMs = new Date().getTime();
+
+export function getNanoTimestamp() {
+	const diffNs = process.hrtime(loadNs);
+	const loadTs = BigInt(loadMs) * 1000000n;
+	const secondsDiff = BigInt(diffNs[0]) * 1000000000n;
+	const nanosDiff = BigInt(diffNs[1]);
+	return loadTs + secondsDiff + nanosDiff;
+}
```

Comment:
You can avoid a lot of repeated calculations here by switching to:
```suggestion
const loadNs = BigInt(Date.now()) * 1000000n - process.hrtime.bigint();

export function getNanoTimestamp() {
	return process.hrtime.bigint() + loadNs;
}
```

Reply:
Nice! was not aware they added a `bigint()`, updated.